packages/db/src/search/StatsSearch.ts

```diff
@@ -0,0 +1,298 @@
+    const knex = sendknex ?? await getKnexClient();
+    const aggregateQuery:Knex.QueryBuilder = knex(`${TableNames.granules}`);
+    if (this.queryStringParameters.timestamp__from) {
+      aggregateQuery.where(`${TableNames.granules}.updated_at`, '>=', new Date(Number.parseInt(this.queryStringParameters.timestamp__from as string, 10)));
+    }
+    if (this.queryStringParameters.timestamp__to) {
+      aggregateQuery.where(`${TableNames.granules}.updated_at`, '<=', new Date(Number.parseInt(this.queryStringParameters.timestamp__to as string, 10)));
+    }
+    aggregateQuery.select(
+      knex.raw(`COUNT(CASE WHEN ${TableNames.granules}.error ->> 'Error' != '{}' THEN 1 END) AS count_errors`),
+      knex.raw(`COUNT(${TableNames.granules}.cumulus_id) AS count_granules`),
+      knex.raw(`AVG(${TableNames.granules}.duration) AS avg_processing_time`),
+      knex.raw(`COUNT(DISTINCT ${TableNames.granules}.collection_cumulus_id) AS count_collections`)
+    );
+    const aggregateQueryRes: TotalSummary[] = await aggregateQuery;
+    return this.formatSummaryResult(aggregateQueryRes[0]);
+  }
+
+  /** Performs joins on the provider and/or collection table if neccessary
+   *
+   * @param {Knex} knex - the knex client to be used
+   * @returns {Knex.QueryBuilder} the knex query of a joined table or not based on queryStringParams
+   */
+  private providerAndCollectionIdBuilder(knex: Knex): Knex.QueryBuilder {
+    let aggregateQuery;
+    this.queryStringParameters.field = this.queryStringParameters.field ? this.queryStringParameters.field : 'status';
+    if (this.queryStringParameters.field?.includes('error.Error')) {
+      aggregateQuery = knex.select(knex.raw(`"count(${typeToTable[this.type]}.cumulus_id), error ->> 'Error' as error"`)).from(typeToTable[this.type]);
+    } else {
+      aggregateQuery = knex.select(`${typeToTable[this.type]}.${this.queryStringParameters.field}`).from(typeToTable[this.type]);
+    }
+    if (this.queryStringParameters.collectionId) {
+      aggregateQuery.join(`${TableNames.collections}`, `${typeToTable[this.type]}.collection_cumulus_id`, 'collections.cumulus_id');
+    }
+
+    if (this.queryStringParameters.provider) {
+      aggregateQuery.join(`${TableNames.providers}`, `${typeToTable[this.type]}.provider_cumulus_id`, 'providers.cumulus_id');
+    }
+    return aggregateQuery;
+  }
+
+  /** Aggregates the search query based on queryStringParameters
+   *
+   * @param {Knex.QueryBuilder} query - the knex query to be aggregated
+   * @param {Knex} knex - the knex client to be used
+   * @returns {Knex.QueryBuilder} the query with its new Aggregatation
+   */
+  private aggregateQueryField(query: Knex.QueryBuilder, knex: Knex): Knex.QueryBuilder {
+    this.queryStringParameters.field = this.queryStringParameters.field ? this.queryStringParameters.field : 'status';
+    if (this.queryStringParameters.field?.includes('error.Error')) {
+      query.select(knex.raw("error ->> 'Error' as error"))
```

Comment:
We can change this `as aggregatedfield`, same for 
query.select(`${typeToTable[this.type]}.${this.queryStringParameters.field} as aggregatedfield`)
groupBy('aggregatedfield')

Reply:
Maybe we can update it in the constructor, so we don't need to check again.
this.queryStringParameters.field = this.queryStringParameters.field?? 'status';